src/Security/MemberAuthenticator/SessionAuthenticationHandler.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace SilverStripe\Security\MemberAuthenticator;
+
+use SilverStripe\Control\Cookie;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Security\Member;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\Session;
+use SilverStripe\Control\Director;
+use SilverStripe\Security\AuthenticationHandler as AuthenticationHandlerInterface;
+use SilverStripe\ORM\ValidationException;
+use SilverStripe\Security\IdentityStore;
+
+/**
+ * Authenticate a member pased on a session cookie
+ */
+class SessionAuthenticationHandler implements AuthenticationHandlerInterface, IdentityStore
+{
+
+    private $sessionVariable;
+
+    /**
+     * Get the session variable name used to track member ID
+     *
+     * @return string
+     */
+    public function getSessionVariable()
+    {
+        return $this->sessionVariable;
+    }
+
+    /**
+     * Set the session variable name used to track member ID
+     *
+     * @param string $sessionVariable
+     * @return null
```

Comment:
`@return $this` preferable for basic setters, or don't declare `@return` at all.